File: SnipeeMac/SnipeeMac/SnipeeMac/Models/HistoryItem.swift
//
//  HistoryItem.swift
//  SnipeeMac
//

import Foundation

struct HistoryItem: Codable, Identifiable {
    let id: String
    var content: String
    var timestamp: Date
    var isPinned: Bool
    
    init(id: String = UUID().uuidString, content: String, timestamp: Date = Date(), isPinned: Bool = false) {
        self.id = id
        self.content = content
        self.timestamp = timestamp
        self.isPinned = isPinned
    }
}

File: app/models/history-item.js
class HistoryItem {
  constructor({
    id = null,
    content = '',
    timestamp = null,
    type = 'history',
    isPinned = false
  } = {}) {
    this.id = id || Date.now().toString();
    this.content = content;
    this.timestamp = timestamp || new Date().toISOString();
    this.type = type;
    this.isPinned = isPinned;
  }
}

module.exports = { HistoryItem };